### PR TITLE
Bug 1142680 - Write a compare-talos equivalent - items 1,2,7,8

### PR DESCRIPTION
**Pull request by jmaher**
[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder-ui/495)

### CODE REVIEW

File: webapp/app/js/compareperf.js [hunk 1]
```diff
@@ -143,8 +144,17 @@ perf.controller('CompareResultsCtrl', [
       });
     }
 
-    var optionCollectionMap = {};
     $scope.dataLoading = true;
+    $scope.getDeltaClass = function(cr) { return cr.className; };
```
Comment (wlach) [inline webapp/app/js/compareperf.js]: I'm not sure about putting CSS logic inside the controller, but it doesn't look like angular provides better options (http://stackoverflow.com/questions/7792652/what-is-the-best-way-to-conditionally-apply-a-class). Could you at least consolidate these into one method like $scope.getCompareResultClasses() ? Just append the applicable classes into a list, then join them into a string and return it.

File: webapp/app/js/compareperf.js [hunk 2]
```diff
@@ -255,6 +278,16 @@ perf.controller('CompareSubtestResultsCtrl', [
       });
     }
 
+    $scope.dataLoading = true;
+    $scope.getDeltaClass = function(cr) { return cr.className; };
```
Comment (wlach) [inline webapp/app/js/compareperf.js]: Code duplication with the above. Maybe we should put the function I suggested above in the compare factory and then make the $scope method a simple wrapper.

File: webapp/app/js/compareperf.js [hunk 3]
```diff
@@ -311,14 +345,24 @@ perf.controller('CompareSubtestResultsCtrl', [
...
+                      }
+                    });
+                  }
+                  if (newSeriesMap) {
+                    Object.keys(newSeriesMap).forEach(function(series) {
+                      if (!_.contains($scope.pageList, newSeriesMap[series].name)) {
+                        $scope.pageList.push(newSeriesMap[series].name);
+                      }
+                    });
+                  }
```
Comment (wlach) [inline webapp/app/js/compareperf.js]: You could put this logic in a forEach that iterates over originalSeriesMap and newSeriesMap

File: webapp/app/js/compareperf.js [hunk 4]
```diff
@@ -326,19 +370,29 @@ perf.controller('CompareSubtestResultsCtrl', [
                   PhSeries.getSubtestSummaries($scope.newProject,
                                 timeRange,
                                 optionCollectionMap,
-                                $scope.originalSignature).then(
+                                $scope.newSignature).then(
                     function (newSeriesData) {
+                      $scope.platformList = _.union($scope.platformList,
+                                                    newSeriesData.platformList).sort();
+                      $scope.testList = _.union($scope.testList,
+                                                    newSeriesData.testList).sort();
```
Comment (wlach) [inline webapp/app/js/compareperf.js]: Nit: indentation

File: webapp/app/js/perf.js [hunk 5]
```diff
@@ -297,6 +334,12 @@ perf.factory('math', [ function() {
       return Math.sqrt(
         values.map(function (v) { return Math.pow(v - avg, 2); })
           .reduce(function (a, b) { return a + b; }) / (values.length - 1));
+    },
+
+    trimFloat: function(number) {
+      if (number === undefined)
+        return "notReady";
```
Comment (wlach) [inline webapp/app/js/perf.js]: This doesn't seem like a great return value. I think we should just return undefined here, and maybe log an error.

File: webapp/app/js/compareperf.js [hunk 6]
```diff
@@ -326,19 +370,29 @@ perf.controller('CompareSubtestResultsCtrl', [
...
                                            timeRange,
                                            [$scope.newResultSetID]);
                   }).then(function(newSeriesMaps) {
                     var newSeriesMap = newSeriesMaps[$scope.newResultSetID];
-                    Object.keys(newSeriesMap).forEach(function(series) {
-                      if (!_.contains($scope.pageList, newSeriesMap[series].name)) {
-                        $scope.pageList.push(newSeriesMap[series].name);
-                      }
-                    });
+                    if (newSeriesMap) {
```
Comment (wlach) [inline webapp/app/js/compareperf.js]: Under what conditions would newSeriesMap not be defined? It kind of looks like we might be papering over something that shouldn't happen?

Comment (jmaher) [inline webapp/app/js/compareperf.js]: updated the PR to address feedback!

On Wed, Apr 29, 2015 at 12:07 PM, William Lachance <notifications@github.com

**Pull request merged**